src/modules/dayCycle/DayCycle.ts

```diff
@@ -2,24 +2,14 @@
 import { Computed } from 'knockout';
 import DayCycleMoment from './DayCycleMoment';
 import DayCyclePart from './DayCyclePart';
+import { DayCycleStartHours } from '../GameConstants';
 import GameHelper from '../GameHelper';
 
 export default class DayCycle {
     public static currentDayCyclePart: Computed<DayCyclePart> = ko.pureComputed(() => {
-        const curHour = GameHelper.currentTime().getHours();
-        let dayCyclePart: DayCyclePart;
+        const currentHour = GameHelper.currentTime().getHours();
 
-        if (curHour === 17) {
-            dayCyclePart = DayCyclePart.Dusk;
-        } else if (curHour === 6) {
-            dayCyclePart = DayCyclePart.Dawn;
-        } else if (curHour > 6 && curHour < 17) {
-            dayCyclePart = DayCyclePart.Day;
-        } else {
-            dayCyclePart = DayCyclePart.Night;
-        }
-
-        return dayCyclePart;
+        return Number(Object.entries(DayCycleStartHours).reverse().find(([, startHour]) => startHour <= currentHour)?.[0] ?? DayCyclePart.Night);
```

Comment:
Just in case Night isn't always the last one,
```suggestion
        return Number(Object.entries(DayCycleStartHours).reverse().find(([, startHour]) => startHour <= currentHour)?.[0] ?? Object.keys(DayCycleStartHours).slice(-1));
```